sanic/request.py

```diff
@@ -347,6 +351,27 @@ def token(self):
 
         return auth_header
 
+    @property
+    def credentials(self) -> Credentials:
+        """Attempt to return the auth header value.
+
+        Covers NoAuth, Basic Auth, Bearer Token, Api Token authentication schemas.
+
+        :return: A named tuple with token or username and password related to request
+        """
+        prefixes = ("Basic", "Bearer", "Token")
+        auth_header = self.headers.getone("authorization", None)
+
+        if auth_header is not None:
+            for prefix in prefixes:
+                if prefix in auth_header:
+                    header_value = auth_header.partition(prefix)[-1].strip()
+                    if prefix == "Basic":
+                        username, password = b64decode(header_value.encode("utf-8")).decode().split(":")
+                        return Credentials(username=username, password=password, token=None)
+                    return Credentials(token=header_value, username=None, password=None)
+        return Credentials(token=auth_header, username=None, password=None)
```

Comment:
We should follow a pattern used elsewhere where we cache the value on a `parsed_*` attribute.

Reply:
Btw, could we switch to Python's `@cached_property` for that? New in Python 3.8... https://docs.python.org/3/library/functools.html#functools.cached_property

Reply:
Absolutely. As soon as we are ready to drop 3.7. Regular support has expired so, we should probably bring this conversation up soon.